### PR TITLE
Abort button

### PR DESCRIPTION
**Pull request by mateusz**

### CODE REVIEW

File: code/control/DNRoot.php [hunk 1]
```diff
@@ -1563,6 +1565,32 @@ public function deploylog(SS_HTTPRequest $request) {
...
 	}
 
+	public function abortDeploy(SS_HTTPRequest $request) {
+		$params = $request->params();
+		$deployment = DNDeployment::get()->byId($params['Identifier']);
+
+		if(!$deployment || !$deployment->ID) {
+			throw new SS_HTTPResponse_Exception('Deployment not found', 404);
+		}
+		if(!$deployment->canView()) {
```
Comment (mateusz) [inline code/control/DNRoot.php]: Needs better security - admin only, or perm code check

File: sass/style.sass [hunk 2]
```diff
@@ -1227,3 +1227,50 @@ textarea.deploy-key
 .content .variables-description p
 	max-width: none
 
+button.abort
```
Comment (halkyon) [inline sass/style.sass]: There's a lot of specific styling here, I reckon you should try to remove it and rely on `.btn` and only add what is absolutely necessary (float: right, etc?) here.

Comment (mateusz) [inline sass/style.sass]: Done - I still want to tweak the size because.

Comment (halkyon) [inline sass/style.sass]: Can it be done with a pre-defined class instead? http://v4-alpha.getbootstrap.com/components/buttons/#sizes

File: javascript/deploynaut.js [hunk 3]
```diff
@@ -275,4 +275,23 @@
 		});
 
 	});
+
+	$('button.abort').click(function(event) {
```
Comment (halkyon) [inline javascript/deploynaut.js]: `on('click', function(event) {` would be better here, as that's recommended over calling click directly.

File: code/control/DNRoot.php (hunk 1 above)
Comment (mateusz) [inline code/control/DNRoot.php]: Done.

File: javascript/deploynaut.js (hunk 3 above)
Comment (mateusz) [inline javascript/deploynaut.js]: ok then I'm not doing it :-)

File: sass/style.sass (hunk 2 above)
Comment (mateusz) [inline sass/style.sass]: Nah, I want it margin-less so it fits inline with the label on the left, and really small, so it doesn't collapse the log. I don't want to fix up the entire page :-| WILL_NOT_PERFORM?

**Comment by mateusz** [comment]
screen shot 2016-07-22 at 4 00 47 pm

File: code/jobs/DeployJob.php [hunk 4]
```diff
@@ -10,7 +10,80 @@ class DeployJob extends DeploynautJob {
...
+			])) {
+				echo sprintf(
+					'[-] Signal "%s" received, delivering to own process group, PID "%s".' . PHP_EOL,
+					$signal,
+					getmypid()
+				);
+				// Mark the signal as received.
+				unlink($sigFile);
+				// Dispatch to own process group.
+				$pgid = posix_getpgid(getmypid());
```
Comment (stojg) [inline code/jobs/DeployJob.php]: double check that the pgid is not zero, bad things can happen since pid zero is the init system.

Comment (mateusz) [inline code/jobs/DeployJob.php]: Fixed

**Pull request merged**